Network/Socket/Syscall.hs

```diff
@@ -135,14 +135,14 @@ bind s sa = withSocketAddress sa $ \p_sa siz -> void $ withFdSocket s $ \fd -> d
 -- Connecting a socket
 
 -- | Connect to a remote socket at address.
-connect :: SocketAddress sa => Socket -> sa -> IO ()
+connect :: (Show sa, SocketAddress sa) => Socket -> sa -> IO ()
 connect s sa = withSocketsDo $ withSocketAddress sa $ \p_sa sz ->
-    connectLoop s p_sa (fromIntegral sz)
+    connectLoop (show sa) s p_sa (fromIntegral sz)
```

Comment:
showed socket addr is lazy right? so no problem with that,
though I think SockAddr Show instance is doing too much and might fail due IO and hide original error.
I would fall back to generated Show and move current Show implementation to a new dedicated method of SocketAddress class. I don't know how much legacy would be affected by such refactoring.

Reply:
@kazu-yamamoto

Reply:
I would like to introduce the `Strict` pragma someday and don't want to relay on the lazy evaluation. So, what about just passing `sa` to `connectLoop` and `connectLoop` calls `show` by itself?